Userland/Applications/Spreadsheet/CellSyntaxHighlighter.cpp

```diff
@@ -13,44 +14,46 @@ namespace Spreadsheet {
 
 void CellSyntaxHighlighter::rehighlight(Palette const& palette)
 {
-    auto text = m_client->get_text();
-    m_client->spans().clear();
-    if (!text.starts_with('=')) {
-        m_client->do_update();
-        return;
-    }
+    m_client->clear_spans();
+
+    if (m_client->get_text().starts_with('=')) {
+
+        JS::SyntaxHighlighter::rehighlight(palette);
+
+        auto spans = m_client->spans();
 
-    JS::SyntaxHighlighter::rehighlight(palette);
-
-    // Highlight the '='
-    m_client->spans().empend(
-        GUI::TextRange { { 0, 0 }, { 0, 1 } },
-        Gfx::TextAttributes {
-            palette.syntax_keyword(),
-            Optional<Color> {},
-            false,
-        },
-        (u64)-1,
-        false);
-
-    if (m_cell && m_cell->thrown_value().has_value()) {
-        if (auto value = m_cell->thrown_value().value(); value.is_object() && is<JS::Error>(value.as_object())) {
-            auto& error = static_cast<JS::Error const&>(value.as_object());
-            auto& traceback = error.traceback();
-            auto& range = traceback.first().source_range;
-            GUI::TextRange text_range { { range.start.line - 1, range.start.column }, { range.end.line - 1, range.end.column } };
-            m_client->spans().prepend(
-                GUI::TextDocumentSpan {
-                    text_range,
+        // Highlight the '='
+        spans.empend(
+            GUI::TextRange { { 0, 0 }, { 0, 1 } },
+            Gfx::TextAttributes {
+                palette.syntax_keyword(),
+                Optional<Color> {},
+                false,
+            },
+            (u64)-1,
+            false);
+
+        if (m_cell && m_cell->thrown_value().has_value()) {
+            if (auto value = m_cell->thrown_value().value(); value.is_object() && is<JS::Error>(value.as_object())) {
+                auto& error = static_cast<JS::Error const&>(value.as_object());
+                auto& range = error.traceback().first().source_range;
+
+                spans.prepend({
+                    GUI::TextRange { { range.start.line - 1, range.start.column }, { range.end.line - 1, range.end.column } },
                     Gfx::TextAttributes {
                         Color::Black,
                         Color::Red,
                         false,
                     },
                     (u64)-1,
-                    false });
+                    false,
+                });
+            }
         }
+
+        m_client->do_set_spans(std::move(spans));
```

Comment:
Drop the `std::` please.

Reply:
Sorry, rookie mistake

Reply:
I think that's a non-rookie best-practice you're following :P 
but nw, it's just a code style